Add jerry-client.py

polaroi8d opened the pull request:
Add the client side of the communication to debugger. 

JerryScript-DCO-1.0-Signed-off-by: Levente Orban orbanl@inf.u-szeged.hu

Commit:
Add jerry-client.py

JerryScript-DCO-1.0-Signed-off-by: Levente Orban orbanl@inf.u-szeged.hu

zherczeg commented on jerry-debugger/jerry-client.py:
```diff
@@ -0,0 +1,39 @@
+        print('Failed to create the socket. Error code: %d, Error message: %s' % (msg[0], msg[1]))
+        sys.exit()
+
+    print ('Socket created on: %d'% PORT)
```
What is the style: space or not space after print?

polaroi8d replied:
Without space, thanks.

stampho commented on jerry-debugger/jerry-client.py:
```diff
@@ -0,0 +1,39 @@
+        client_socket.connect((HOST, PORT))
+    except socket.error, msg:
+        print('Failed to create the socket. Error code: %d, Error message: %s' % (msg[0], msg[1]))
+        sys.exit()
```
Return code is missing from here. Probably use sys.exit(1) or you might use the error code above but not sure whether is this the common way to do this.

Hint: you can easily check the sys.exit() usage in the Jerry's python scripts with the following command:
find . -name "*.py" | xargs grep --color "sys.exit"

polaroi8d replied:
Thanks for the instruction.

stampho commented on jerry-debugger/jerry-client.py:
```diff
@@ -0,0 +1,39 @@
+        print('Failed to create the socket. Error code: %d, Error message: %s' % (msg[0], msg[1]))
+        sys.exit()
+
+    print ('Socket created on: %d'% PORT)
```
nitpick: style issues with space:
1. print (
2. %d'%

polaroi8d replied:
:+1:

stampho commented on jerry-debugger/jerry-client.py:
```diff
@@ -0,0 +1,39 @@
+        client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        client_socket.connect((HOST, PORT))
+    except socket.error, msg:
+        print('Failed to create the socket. Error code: %d, Error message: %s' % (msg[0], msg[1]))
```
one more thing: probably you should send error message to the stderr. Check sys.stderr.write in the python documentation.

polaroi8d replied:
Okey.

Commit:
Add stderr print
 - Fix style issues

JerryScript-DCO-1.0-Signed-off-by: Levente Orban orbanl@inf.u-szeged.hu

stampho commented on jerry-debugger/jerry-client.py:
```diff
@@ -0,0 +1,39 @@
+        client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        client_socket.connect((HOST, PORT))
+    except socket.error, msg:
+        print >> sys.stderr, ('Failed to create the socket. Error code: %d, Error message: %s' % (msg[0], msg[1]))
```
Most probably this won't work with python3. I would propose to use sys.stderr.write('message\n')

Anyway, I've just found something: http://stackoverflow.com/questions/2949974/how-to-exit-a-program-sys-stderr-write-or-print

It seems sys.exit('message') is a proposed way and exits with exit code 1

Commit:
Fix the python3 error and add strerr

stampho commented:
LGTM

zherczeg commented:
LGTM
